x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -1372,6 +1376,66 @@ class AgentPolicyService {
     return { updatedPolicies: updatedPoliciesSuccess, failedPolicies };
   }
 
+  public fetchAllAgentPolicyIds(
+    soClient: SavedObjectsClientContract,
+    { perPage = 1000, kuery = undefined }: FetchAllAgentPolicyIdsOptions = {}
+  ): AsyncIterable<string[]> {
+    return createSoFindIterable<{}>({
+      soClient,
+      findRequest: {
+        type: SAVED_OBJECT_TYPE,
+        perPage,
+        sortField: 'created_at',
+        sortOrder: 'asc',
+        fields: ['id'],
+        filter: kuery ? normalizeKuery(SAVED_OBJECT_TYPE, kuery) : undefined,
+      },
+      resultsMapper: (data) => {
+        return data.saved_objects.map((agentPolicySO) => {
+          auditLoggingService.writeCustomSoAuditLog({
+            action: 'find',
+            id: agentPolicySO.id,
+            savedObjectType: AGENT_POLICY_SAVED_OBJECT_TYPE,
+          });
+          return agentPolicySO.id;
+        });
+      },
+    });
+  }
+
+  public fetchAllAgentPolicies(
+    soClient: SavedObjectsClientContract,
+    {
+      perPage = 1000,
+      kuery,
+      sortOrder = 'asc',
+      sortField = 'created_at',
+      fields = [],
+    }: FetchAllAgentPoliciesOptions = {}
+  ): AsyncIterable<AgentPolicy[]> {
+    return createSoFindIterable<AgentPolicySOAttributes>({
+      soClient,
+      findRequest: {
+        type: SAVED_OBJECT_TYPE,
+        sortField,
+        sortOrder,
+        perPage,
+        fields,
+        filter: kuery ? normalizeKuery(SAVED_OBJECT_TYPE, kuery) : undefined,
+      },
+      resultsMapper(data) {
+        return data.saved_objects.map((agentPolicySO) => {
+          auditLoggingService.writeCustomSoAuditLog({
+            action: 'find',
+            id: agentPolicySO.id,
+            savedObjectType: AGENT_POLICY_SAVED_OBJECT_TYPE,
+          });
+          return { id: agentPolicySO.id, ...agentPolicySO.attributes };
```

Comment:
Can you see if Fleet already has a "mapper" for converting the SO attributes to `AgentPolicy` structure?